src/GeneralStateTestsFiller/stReturnDataTest/clearReturnBufferFiller.yml

```diff
@@ -0,0 +1,518 @@
+           function makeFailConstructor(addr, len) -> retVal {
+              let endOfGood := makeConstructor(addr, len)
+              
+              // Replace the STOP with 0xFE, which is the designated bad opcode
+              mstore8(sub(endOfGood, 1), 0xFE)
+              retVal := endOfGood
+           }
+
+
+           switch operation
+           case 0xF0 {  // CREATE
+              codeLen := makeConstructor(bufferFrom, bufLen)
+              pop(create(0, 0, codeLen))
+           }
+           case 0x11F0 {  // CREATE, which fails
+              codeLen := makeFailConstructor(bufferFrom, bufLen)
+              pop(create(0, 0, codeLen))
+           }
+           case 0xF5 {  // CREATE2
+              codeLen := makeConstructor(bufferFrom, bufLen)
+              pop(create2(0, 0, codeLen, 0x5a17))
+           }
+           case 0x11F5 {  // CREATE2, which fails
+              codeLen := makeFailConstructor(bufferFrom, bufLen)
+              pop(create2(0, 0, codeLen, 0x5a17))
+           }
+           case 0xF1 { // CALL
+              mstore(0x00, bufferFrom)
+              mstore(0x20, bufLen)
+              pop(call(gas(), 0x00ca11ed, 0, 0, 0x40, 0x100, 0x20))
+           }
+           case 0x11F1 { // CALL, but it fails
+              mstore(0x00, bufferFrom)
+              mstore(0x20, bufLen)
+              pop(call(gas(), 0xbad0ca11, 0, 0, 0x40, 0x100, 0x20))
+           }
+           case 0xF2 { // CALLCODE
+              mstore(0x00, bufferFrom)
+              mstore(0x20, bufLen)
+              pop(callcode(gas(), 0x00ca11ed, 0, 0, 0x40, 0x100, 0x20))
+           }
+           case 0x11F2 { // CALLCODE, but it fails
+              mstore(0x00, bufferFrom)
+              mstore(0x20, bufLen)
+              pop(callcode(gas(), 0xbad0ca11, 0, 0, 0x40, 0x100, 0x20))
+           }
+           case 0xF4 { // DELEGATECALL
+              mstore(0x00, bufferFrom)
+              mstore(0x20, bufLen)
+              pop(delegatecall(gas(), 0x00ca11ed, 0, 0x40, 0x100, 0x20))
+           }
+           case 0x11F4 { // DELEGATECALL, but it fails
+              mstore(0x00, bufferFrom)
+              mstore(0x20, bufLen)
+              pop(delegatecall(gas(), 0xbad0ca11, 0, 0x40, 0x100, 0x20))
+           }
+           case 0xFA { // STATICCALL
+              mstore(0x00, bufferFrom)
+              mstore(0x20, bufLen)
+              pop(staticcall(gas(), 0x00ca11ed, 0, 0x40, 0x100, 0x20))
+           }
+           case 0x11FA { // STATICCALL, but it fails
+              mstore(0x00, bufferFrom)
+              mstore(0x20, bufLen)
+              pop(staticcall(gas(), 0xbad0ca11, 0, 0x40, 0x100, 0x20))
+           }
+           default {  // We shouldn't get here
+             revert(0,0)
+           }
+           // End of switch operation
+
+           sstore(0, returndatasize())    // Should be zero
+           // We can't do returndatacopy() here because it causes the transaction 
```

Comment:
returndatacopy of returndatasize cause revert? do we have a test about it?

Reply:
Yes, it does. Here's the `vmtraceraw` line that shows it:

```
{"pc":673,"op":62,"gas":"0x12fbde","gasCost":"0x6","memory":"0x","memSize":96,"stack":["0x11f0","0xf3f3","0x20","0x5c","0x20","0x0","0x0"],"returnData":"0x","depth":1,"refund":4800,"opName":"RETURNDATACOPY","error":""}
{"output":"","gasUsed":"0x4c45ffc","time":2642639,"error":"return data out of bounds"}
```

> do we have a test about it?

I'll write that test.

Reply:
#1010 is the PR for the test